### PR TITLE
Removed PowerMock and the 2 unit tests using it

### PR DESCRIPTION
**Pull request by dlmarion**
Removed CompactionCoordinatorTest and CompactorTest which were testing the state of the CompactionCoordinator and Compactor internals. Removing these tests allowed me to remove PowerMock and JUnit4 from the build.

### CODE REVIEW

**Comment by ctubbsii** [comment]
Why do this? Didn't we want this test coverage?

**Review by ctubbsii — CHANGES_REQUESTED** [review]
src/build/ci/find-unapproved-junit.sh was also involved in checking the use of JUnit during a build.

File: pom.xml [hunk 1]
```diff
@@ -981,8 +954,6 @@
                 <unused>org.apache.logging.log4j:log4j-1.2-api:jar:*</unused>
                 <unused>org.apache.logging.log4j:log4j-slf4j2-impl:jar:*</unused>
                 <unused>org.apache.logging.log4j:log4j-web:jar:*</unused>
-                <!-- This should be removed upon completion of migrating junit 4 to 5 -->
-                <unused>org.junit.vintage:junit-vintage-engine:jar:*</unused>
```
Comment (DomGarguilo) [inline pom.xml:985]: I think there is also an entry in the `ignoredUsedUndeclaredDependencies` section that can be removed like this one.

Comment (DomGarguilo) [reply pom.xml:985]: Additionally in the pom, in the `maven-checkstyle-plugin` section, we might be able to remove some rules that refference junit4 imports or add one/some that ensure only junit jupiter imports are used.

**Comment by dlmarion** [comment]
> Why do this? Didn't we want this test coverage?

As I mentioned in the JDK 21 PR, I don't think these PowerMock tests are going to be compatible with Java 21 and I believe that we have sufficient coverage.

**Comment by ctubbsii** [comment]
> I believe that we have sufficient coverage.

Okay, that's the part I was concerned about. If we have sufficient coverage, I'm okay with moving forward on dropping these, so long as we get all the little remaining bits included.

File: pom.xml (hunk 1 above)
Comment (keith-turner) [reply pom.xml:985]: So removing powermock mean we can remove usage of all old junit apis?  If so that is nice, it would be nice if the old stuff was no longer there in autocomplete in IDE.

**Review by EdColeman — COMMENTED** [review]
The changes look okay - need to address @DomGarguilo comment with additional things that may be able to be removed along with these changes.

**Comment by dlmarion** [comment]
I re-reviewed the tests that are being removed. The CompactionCoordinatorTest is checking the internal state of the `RUNNING_CACHE` and `QUEUE_SUMMARIES` objects in the CompactionCoordinator in various conditions. The CompactorTest is confirming that `updateCompactionFailed`, `updateCompactionCompleted` and `updateCompactionState` in the Compactor are called.

The ITs certainly cover the calls to `updateCompactionFailed`, `updateCompactionCompleted` in the CompactorTest. Calls to `updateCompactionState` are not critical as that just relays that progress is happening. I don't know that the CompactionCoordinator tests are comprehensive enough to cover all edge cases for the internal state.

I've re-conviced myself that I'm fine with removing these tests.

File: server/compaction-coordinator/src/test/java/org/apache/accumulo/coordinator/CompactionCoordinatorTest.java [hunk 2]
```diff
@@ -1,657 +0,0 @@
...
-    PowerMock.replayAll();
-
-    var coordinator = new TestCoordinator(finalizer, tservers, client, tsc, context, security);
-    coordinator.resetInternals();
-    assertEquals(0, coordinator.getQueues().size());
-    assertEquals(0, coordinator.getIndex().size());
-    assertEquals(0, coordinator.getRunning().size());
-    coordinator.run();
-    assertEquals(1, coordinator.getQueues().size());
-    QueueAndPriority qp = QueueAndPriority.get("R2DQ".intern(), (short) 1);
```
Comment (keith-turner) [inline server/compaction-coordinator/src/test/java/org/apache/accumulo/coordinator/CompactionCoordinatorTest.java:296]: we are losing `R2DQ`

File: src/build/ci/find-unapproved-junit.sh [hunk 3]
```diff
@@ -38,8 +36,10 @@ function findalljunitproblems() {
   fi
   # find any new classes using something other than the jupiter API, except those allowed
   grep "$opts" --include='*.java' 'org[.]junit[.](?!jupiter)' | grep -Pv "^(${ALLOWED_PIPE_SEP//./[.]})\$"
-  # find any uses of the jupiter API in the allowed vintage classes
-  grep "$opts" 'org[.]junit[.]jupiter' "${ALLOWED[@]}"
+  if ((${#ALLOWED[@]} != 0)); then
```
Comment (keith-turner) [inline src/build/ci/find-unapproved-junit.sh:39]: Could be a follow on issue, wondering if the ALLOWED set should just be removed from the script since its empty.  Would simplify the script.

**Review by keith-turner — APPROVED** [review]

File: src/build/ci/find-unapproved-junit.sh (hunk 3 above)
Comment (dlmarion) [reply src/build/ci/find-unapproved-junit.sh:39]: I left it there just in case we needed it in the future for some reason. We have the same thing in accumulo-access with this script.

File: pom.xml [hunk 4]
```diff
@@ -1092,26 +1055,14 @@
                   <property name="format" value="org[.]apache[.]commons[.]math[.]" />
                   <property name="message" value="Use commons-math3 (org.apache.commons.math3.*)" />
                 </module>
-                <module name="RegexpSinglelineJava">
-                  <property name="format" value="junit[.]framework[.]TestCase" />
-                  <property name="message" value="Use JUnit4+ @Test annotation instead of TestCase" />
-                </module>
```
Comment (ctubbsii) [inline pom.xml:1098]: I don't think this line should have been removed. This was checking to make sure we were *not* using JUnit 3. Instead of removing this, we should have added the checkstyle to make sure we were *not* using JUnit 3 **or** 4. Then, we could get rid of the junit script that was being run during GitHub Actions, as checkstyle would have covered it.

**Comment by cshannon** [comment]
This PR has broken the sunny tests, I now get the following when running -Psunny:

`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:3.1.2:integration-test (run-integration-tests) on project accumulo-start: groups/excludedGroups require TestNG, JUnit48+ or JUnit 5 (a specific engine required on classpath) on project test classpath -> [Help 1]`

I'm taking a look now on how to fix it.

**Comment by cshannon** [comment]
> This PR has broken the sunny tests, I now get the following when running -Psunny:
> 
> `[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:3.1.2:integration-test (run-integration-tests) on project accumulo-start: groups/excludedGroups require TestNG, JUnit48+ or JUnit 5 (a specific engine required on classpath) on project test classpath -> [Help 1]`
> 
> I'm taking a look now on how to fix it.

Opened https://github.com/apache/accumulo/pull/3829 to fix this
